notebook/autobuild_function_calling.ipynb

```diff
@@ -0,0 +1,405 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import autogen\n",
+    "from autogen.agentchat.contrib.agent_builder import AgentBuilder\n",
+    "\n",
+    "config_file_or_env = \"OAI_CONFIG_LIST\"\n",
+    "config_list = autogen.config_list_from_json(config_file_or_env, filter_dict={\"model\": [\"gpt-4-1106-preview\", \"gpt-4\"]})\n",
+    "llm_config = {\n",
+    "    \"config_list\": config_list,\n",
+    "    \"timeout\": 120,\n",
+    "}\n",
+    "\n",
+    "\n",
+    "def start_task(execution_task: str, agent_list: list):\n",
+    "    group_chat = autogen.GroupChat(agents=agent_list, messages=[], allow_repeat_speaker=agent_list[:-1], max_round=12)\n",
+    "    manager = autogen.GroupChatManager(groupchat=group_chat, llm_config={\"config_list\": config_list})\n",
+    "    agent_list[0].initiate_chat(manager, message=execution_task)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 2: Create a AgentBuilder\n",
+    "\n",
+    "Create a `AgentBuilder` with the specified `config_path_or_env`. AgentBuilder will use `gpt-4` in default to complete the whole process, you can specify the `builder_model` and `agent_model` to other OpenAI model to match your task. You can also specify an open-source LLM supporting by vLLM and FastChat, see blog for more details."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "builder = AgentBuilder(\n",
+    "    config_file_or_env=config_file_or_env, builder_model=\"gpt-4-1106-preview\", agent_model=\"gpt-4-1106-preview\"\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 3: Specify a building task\n",
+    "\n",
+    "Specify a building task with a general description. Building task will help build manager (a LLM) decide what agents should be built."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "building_task = \"Get arxiv papers published in last 24 hours\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 4: Define functions\n",
+    "\n",
+    "Define functions to be executed by the Agents of AutoBuild, further specify details like `name`, `description` and `function` of all the functions in an array called `list_of_functions` which will be passed to `builder.build()`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import time\n",
+    "from datetime import datetime, timedelta\n",
+    "from typing import Dict\n",
+    "\n",
+    "import feedparser\n",
+    "\n",
+    "\n",
+    "def get_arxiv_papers() -> Dict:\n",
```

Comment:
The tool's name should reflect its features. `get_arxiv_paper_from_yesterday` is better.